### PR TITLE
feat: show upgrade dialog on get-starknet calls

### PR DESCRIPTION
**Pull request by khanti42**
This PR is to add upgrade dialog on below APIs where those are using by get-starknet:

- `starkNet_signDeclareTransaction`
- `starkNet_signDeployAccountTransaction`
- `starkNet_executeTxn`
- `starkNet_declareContract`

if above APIS trigger with an Cairo 0, account contract, it will first display an dialog, then reject the request by throw an Error

image

### CODE REVIEW

File: packages/starknet-snap/src/declareContract.ts [hunk 1]
```diff
@@ -16,6 +16,11 @@ export async function declareContract(params: ApiParams) {
     const senderAddress = requestParamsObj.senderAddress;
     const network = getNetworkFromChainId(state, requestParamsObj.chainId);
     const { privateKey } = await getKeysFromAddress(keyDeriver, network, state, senderAddress);
+    
+    if(isUpgradeRequired(network, senderAddress)){
+      showUpgradeRequestModal(wallet);
+      return false;
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/declareContract.ts:22]: perhaps throw an error is better?
in terms of end user exp, we handle by dialog
in terms of the coding handle, return false may not fit to the purpose for the caller to determine it is success or not , throw an error seem more general 

WDYT

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]

File: packages/starknet-snap/src/declareContract.ts (hunk 1 above)
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/declareContract.ts:22]: BTW , test is not pass

Comment (stanleyyconsensys) [reply packages/starknet-snap/src/declareContract.ts:22]: you have to await for showUpgradeRequestModal

**Comment by khanti42** [comment]
I already merged latest changes from main (including the deployment check for executeTxn), the PR#254 including these changes in release/2.3.0 is still open

**Review by stanleyyconsensys — APPROVED** [review]
LGTM

**Review by stanleyyconsensys — APPROVED** [review]

**Pull request merged**